requests_ntlm/requests_ntlm.py

```diff
@@ -104,18 +104,15 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
             if s.startswith(auth_strip)
         ).strip()
 
-        ServerChallenge, NegotiateFlags = ntlm.parse_NTLM_CHALLENGE_MESSAGE(
-            ntlm_header_value[len(auth_strip):]
-        )
+        # Parse the challenge in the ntlm context
+        context.parse_challenge_message(ntlm_header_value[len(auth_strip):])
 
         # build response
-
-        # ntlm returns the headers as a base64 encoded bytestring. Convert to a
-        # string.
-        auth = '%s %s' % (auth_type, ntlm.create_NTLM_AUTHENTICATE_MESSAGE(
-            ServerChallenge, self.username, self.domain, self.password,
-            NegotiateFlags
-        ).decode('ascii'))
+        # Get the response based on the challenge message
+        authenticate_message = context.create_authenticate_message(self.username, self.password, self.domain,
+                                                                        server_certificate_hash=server_certificate_hash)
+        authenticate_message = authenticate_message.decode('ascii')
+        auth = '%s %s' % (auth_type, authenticate_message)
```

Comment:
I think it would be better to have this be an explicit unicode string to avoid casting on Python 2. Just add a `u` prefix and that'll cover it.